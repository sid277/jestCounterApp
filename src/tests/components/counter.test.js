import React from 'react';
import { shallow} from 'enzyme';
import Counter from '../../components/counter';

describe("Test the Counter file",()=>{

  const data = { id: 1, value: 2 };

  test('should test snapshot of Counter component', () => {
    const wrapper = shallow(<Counter counter={data}/>);
    expect(wrapper).toMatchSnapshot();
    });

  test('should test the value showing in Counter component', () => {
    const wrapper = shallow(<Counter key={data.id} counter={data}/>);
    const displayValue = wrapper.find('span');
    expect(displayValue.text()).toBe("2");
    });

  test('test the increment button', () => {
    let mockFn = jest.fn();
    const wrapper = shallow(<Counter key={data.id} counter={data} onIncrement={mockFn}/>);
    wrapper.find('[className="btn btn-secondary"]').simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  test('test the decrement button', () => {
    let mockFn = jest.fn();
    const wrapper = shallow(<Counter key={data.id} counter={data} onDecrement={mockFn}/>);
    wrapper.find('[className="btn btn-info m-2"]').simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  test('test the decrement button is disabled on counter.value ==0',()=>{
    let data1 = {id:1, value:0};
    const wrapper = shallow(<Counter key={data1.id} counter={data1}/>);
    const decrementBtn = wrapper.find('[className="btn btn-info m-2"]');
    expect(decrementBtn).toBeDisabled();
  });

  test('test the delete button', () => {
    let mockFn = jest.fn();
    const wrapper = shallow(<Counter key={data.id} counter={data} onDelete={mockFn}/>);
    wrapper.find('[className="btn btn-danger"]').simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  test('test the classname of span tag', () => {
    const wrapper = shallow(<Counter key={data.id} counter={data}/>);
    expect(wrapper.find('span').hasClass('badge m-2 badge-primary')).toBe(true);
  });

});