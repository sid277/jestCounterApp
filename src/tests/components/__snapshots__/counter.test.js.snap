// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Counter file should test snapshot of Counter component 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-1"
    >
      <span
        className="badge m-2 badge-primary"
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        2
      </span>
    </div>
    <div
      className="col-md-4"
    >
      <button
        className="btn btn-secondary"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-plus-circle"
        />
      </button>
      <button
        className="btn btn-info m-2"
        disabled=""
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-minus-circle"
        />
      </button>
      <button
        className="btn btn-danger"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-trash-o"
        />
      </button>
    </div>
  </div>
</div>
`;
